examples/plotting/time_distance_image.py

```diff
@@ -0,0 +1,92 @@
+"""
+=====================================================
+Creating a time-distance plot from a sequence of maps
+=====================================================
+
+This example showcases how you can use :func:`sunpy.map.pixelate_coord_path` and
+:func:`sunpy.map.sample_at_coords()` on a sequence of images to create a time-distance diagram.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.coordinates import propagate_with_solar_surface
+from sunpy.coordinates.screens import SphericalScreen
+from sunpy.net import Fido
+from sunpy.net import attrs as a
+
+###############################################################################
+# First, we will need to acquire a sequence of images from SDO/AIA.
+
+query = Fido.search(
+    a.Time('2025-05-01 00:00:00', '2025-05-01 00:10:00'),
+    a.Instrument.aia,
+    a.Wavelength(171*u.angstrom)
+)
+files = Fido.fetch(query)
+aia_sequence = sunpy.map.Map(files, sequence=True)
+
+###############################################################################
+# Our target will be a set of loops in the corona over a small bright region.
+# We will create a rectangular slice (i.e., a map cutout) around this region.
+
+corner = SkyCoord(Tx=800*u.arcsec, Ty=200*u.arcsec, frame=aia_sequence[0].coordinate_frame)
+cut_sequence = []
+for aia_map in aia_sequence:
+    with SphericalScreen(aia_map.observer_coordinate):
+        cut_sequence.append(aia_map.submap(corner, width=400*u.arcsec, height=400*u.arcsec))
+cut_aia_sequence = sunpy.map.Map(cut_sequence, sequence=True)
+
+###############################################################################
+# Now that we have our cutouts, we can reproject each map in our sequence to
+# the WCS of that cutout using the `~sunpy.coordinates.propagate_with_solar_surface`
+# context manager to adjust the field of view of the cutout with the rotation of the solar surface.
+
+with propagate_with_solar_surface():
+    with SphericalScreen(cut_aia_sequence[0].observer_coordinate):
```

Comment:
Is this correct?

Reply:
Nope, you generally can't combine the two context managers (xref #7173)

Reply:
This feature got added, so I think we're ok.